lib/rspec/support/method_signature_verifier.rb

```diff
@@ -365,12 +365,12 @@ def unlimited_args?
       def split_args(*args)
         kw_args = if @signature.has_kw_args_in?(args)
                     last = args.pop
-                    non_kw_args = last.reject { |k, _| k.is_a?(Symbol) }
+                    non_kw_args = last.reject { |k, _| k.is_a?(Symbol) || k.is_a?(String) }
                     if non_kw_args.empty?
                       last.keys
                     else
                       args << non_kw_args
-                      last.select { |k, _| k.is_a?(Symbol) }.keys
+                      last.select { |k, _| k.is_a?(Symbol) || k.is_a(String) }.keys
```

Comment:
Question mark is missing.
This makes me think this branch is not covered in tests. Can you please check this?

Reply:
I think the whole chunk of code is unnecessary for the current ruby version.  I went through the [PR](https://github.com/rspec/rspec-support/pull/366) that added it, and the parameter declarations that PR was written to support no longer function (at least in my irb console which is ruby 3).

Example:
```
def foo(x, y = {}, z: 1) puts "#{y}, #{z}"; end
foo 1, 'a' => 2, 'b' => 2
ArgumentError (unknown keywords: "a", "b")
```

So i'm not sure what the correct thing to do here is.  Do I update the code to respect ruby 3 functionality?  Or does there need to be some if statement on the runtime version in the code?

Reply:
We can conditionally define methods depending on the Ruby version/features.

Does the “any key type counts as an optional kwargs key” apply to all Ruby versions that suport optional kwargs?

If there’s a conditional branch that isn’t used as we’ve discovered, let’s double check why it was introduced initially. I’d remove if it’s dead code.

Reply:
This section was re-written so these comments are a bit outdated.

But in reference to what you are asking, yes the logic needs to stay for ruby < 3, and the proposed change is correct for ruby >= 3.  The latest version of this PR introduces a RubyFeatures method to allow for this, and a link to the ruby changelog explaining the change.